Isabelle/Chi_Calculus/Relaying.thy

```diff
@@ -40,7 +40,7 @@ proof -
   have "a \<leftrightarrow> b \<parallel> b \<rightarrow> a \<sim>\<^sub>\<flat> b \<leftrightarrow> a \<parallel> b \<rightarrow> a"
     using basic.bisimilarity_transitivity_rule parallel_associativity parallel_commutativity by blast
   also have "b \<leftrightarrow> a \<parallel> b \<rightarrow> a \<sim>\<^sub>\<flat> b \<leftrightarrow> a"
-    by (simp add: forward_bridge_absorption)
+    using forward_bridge_absorption by blast
```

Comment:
In this particular case, you can actually replace `by blast` with `.`.

However, why doesn’t `simp` work anymore? Because `distributor` uses `∏`, and `∏`-terms are expanded by `simp` since `general_parallel.simps` is part of the simpset (you can see this effect by looking at the goal left over by `simp`). The unexpanded left-hand side of `forward_bridge_absorption` doesn’t match the expanded left-hand side of the goal. If we expand `forward_bridge_absorption` as well, `simp` works again. We can expand it by making it a chained fact; so `using forward_bridge_absorption by simp` works.

That said, the proper way to prove such goal these days is to use the `equivalence` method. Could you please check whether using `equivalence` works in the other places where the former `simp` applications don’t work anymore?

For the proof above we shouldn’t use `equivalence` though, since, as I wrote above, a simple `.` is enough.